tests/webTables.spec.ts

```diff
@@ -225,133 +136,85 @@ test('Validate specialty update', async ({ page }) => {
-    // Navigate to the 'Veterinarians' page through the dropdown menu at the top
-    await page.getByText("Veterinarians").click();
+     // Navigate to the 'Veterinarians' page through the dropdown menu at the top
+     
+    await page.getByText("Veterinarians").click()
     await page.getByText("All").click()
 
     /* 10. On the Veterinarians page, add assertion that "Rafael Ortega" has specialty "dermatology" */
-    // Iterate through each row of the table
-    for (const row of allRows) {
-        // Collect each cell in the row
-        const rowData = row.getByRole('cell')
-
-        // Make sure the row has cells
-        if((await rowData.all()).length > 0){
-
-            // Find Raphael Ortega specifically
-            if (await rowData.nth(0).textContent() === 'Raphael Ortega') {
-
-                // Validate that Raphael's specialty is 'surgery'
-                await expect(rowData.nth(1)).toHaveText('dermatology')
-            }
-        }
-    }
+    // Validate Raphael's veterinarian specialty
+    await expect(page.getByRole('row', {name: 'Rafael Ortega'}).getByRole('cell').nth(1)).toHaveText('dermatology')
 
     /* 11. Navigate to SPECIALTIES page, revert the changes renaming "dermatology" back to "surgery" */
-
     // Navigate to the Specialties page
     await page.getByRole('link', {name: 'Specialties'}).click()
-    
-    // Wait for all table contents to load
-    await page.waitForTimeout(timeout)
-
-    // Iterate through the table rows to update 'surgery' to 'dermatology'
-    for(const row of allSpecialtyRows){
-
-        // Collect each cell in the row
-        const rowData = await row.getByRole('textbox').all()
-
-        // Skipping the header
-        if(rowData.length > 0){
 
-            // Locate the row containing 'dermatology', which would be in the first cell
-            if(await rowData[0].inputValue() === 'dermatology'){
-            
-                // Find the Edit button and click it to edit the record
-                await row.getByText('Edit').click()
-                await page.waitForTimeout(timeout)
-
-                // Update 'dermatology' specialty back to 'surgery'
-                await page.getByRole('textbox').clear()
-                await page.getByRole('textbox').fill('surgery')
-                await page.getByText('Update').click()
-
-                // Exit loop
-                break;
-            }
+   // Iterate through rows to find 'dermatology'
+   for (let row of allRows) {
+        // If 'surgery' is found...
+        if( await row.locator('input').inputValue() === 'dermatology'){
+            // ... click 'Edit'
+            await row.getByRole('button', { name: 'Edit' }).click()
+            break;
         }
     }
+
+    // Await 'surgery' to load in the input...
+    await expect(page.locator('#name')).toHaveValue('dermatology')
+
+    // Revert dermatology back to surgery
+    await page.locator('#name').clear()
+    await page.locator('#name').fill('surgery')
```

Comment:
This locator repeated in the code 6 times. It worth to extract it to a constant.